### PR TITLE
[GLUTEN-6159][CH] Support `array_sort`

### PR DESCRIPTION
**Pull request by lgbo-ustc**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

Fixes: #6159

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: cpp-ch/local-engine/Functions/SparkFunctionSortArray.h [hunk 1]
```diff
@@ -69,14 +69,14 @@ struct SparkArraySortImpl
 
 struct NameArraySort
```
Comment (exmy) [inline cpp-ch/local-engine/Functions/SparkFunctionSortArray.h:70]: Better rename to `NameSortArray`

File: cpp-ch/local-engine/Functions/SparkFunctionSortArray.h [hunk 2]
```diff
@@ -69,14 +69,14 @@ struct SparkArraySortImpl
...
 {
-    static constexpr auto name = "arraySortSpark";
+    static constexpr auto name = "sortArraySpark";
 };
 struct NameArrayReverseSort
```
Comment (exmy) [inline cpp-ch/local-engine/Functions/SparkFunctionSortArray.h:74]: Better rename to `NameReverseSortArray`

File: cpp-ch/local-engine/Parser/SerializedPlanParser.cpp [hunk 3]
```diff
@@ -117,15 +117,15 @@ std::string join(const ActionsDAG::NodeRawConstPtrs & v, char c)
...
+    //if (logger->debug())
     {
         namespace pb_util = google::protobuf::util;
         pb_util::JsonOptions options;
         std::string json;
         auto s = pb_util::MessageToJsonString(message, &json, options);
         if (!s.ok())
             throw Exception(ErrorCodes::LOGICAL_ERROR, "Can not convert {} to Json", type);
-        LOG_DEBUG(logger, "{}:\n{}", type, json);
+        LOG_ERROR(logger, "{}:\n{}", type, json);
```
Comment (exmy) [inline cpp-ch/local-engine/Parser/SerializedPlanParser.cpp:128]: keep `LOG_DEBUG`?

**Comment by zhanglistar** [comment]
LGTM

**Review by exmy — APPROVED** [review]

**Pull request merged**